otherlibs/threads/pervasives.ml

```diff
@@ -553,6 +624,18 @@ let close_in_noerr ic = (try close_in ic with _ -> ());;
 external set_binary_mode_in : in_channel -> bool -> unit
                             = "caml_ml_set_binary_mode"
 
+let with_open_in_gen mode perm name f =
+  let c = open_in_gen mode perm name in
+  finally_ f close_in_noerr c
```

Comment:
In case of success of f, an exception raised by close_in would be hidden.

Why the asymmetry with the out functions?